core/src/test/scala/sbt/librarymanagement/CrossVersionTest.scala

```diff
@@ -284,6 +284,55 @@ class CrossVersionTest extends UnitSpec {
     patchVersion("2.11.8-X1.5-bin-extra") shouldBe Some("artefact_2.11.8-X1.5")
   }
 
+  "isScalaBinaryCompatibleWith" should "for (2.10.4, 2.10.5) return true" in {
+    isScalaBinaryCompatibleWith("2.10.4", "2.10.5") shouldBe true
+  }
+  it should "for (2.10.6, 2.10.5) return true" in {
+    isScalaBinaryCompatibleWith("2.10.6", "2.10.5") shouldBe true
+  }
+  it should "for (2.11.0, 2.10.5) return false" in {
+    isScalaBinaryCompatibleWith("2.11.0", "2.10.5") shouldBe false
+  }
+  it should "for (3.0.0, 2.10.5) return false" in {
+    isScalaBinaryCompatibleWith("3.0.0", "2.10.5") shouldBe false
+  }
+  it should "for (3.0.0, 3.1.0) return false" in {
+    isScalaBinaryCompatibleWith("3.0.0", "3.1.0") shouldBe false
+  }
+  it should "for (3.1.0, 3.0.0) return true" in {
+    isScalaBinaryCompatibleWith("3.1.0", "3.0.0") shouldBe true
+  }
+  it should "for (3.1.0, 3.1.1) return true" in {
+    isScalaBinaryCompatibleWith("3.1.0", "3.1.1") shouldBe true
+  }
+  it should "for (3.1.1, 3.1.0) return true" in {
+    isScalaBinaryCompatibleWith("3.1.1", "3.1.0") shouldBe true
+  }
+  it should "for (2.10.0-M1, 2.10.5) return true" in {
+    isScalaBinaryCompatibleWith("2.10.0-M1", "2.10.5") shouldBe true
+  }
+  it should "for (2.10.5, 2.10.0-M1) return true" in {
+    isScalaBinaryCompatibleWith("2.10.5", "2.10.0-M1") shouldBe true
+  }
+  it should "for (2.10.0-M1, 2.10.0-M2) return true" in {
+    isScalaBinaryCompatibleWith("2.10.0-M1", "2.10.0-M2") shouldBe true
+  }
```

Comment:
Non-release versions probably shouldn't be compatible.

https://github.com/sbt/sbt/pull/6939 failed on https://github.com/sbt/sbt/blob/1.7.x/sbt-app/src/sbt-test/actions/cross-strict-aggregation

Reply:
https://github.com/sbt/librarymanagement/pull/401